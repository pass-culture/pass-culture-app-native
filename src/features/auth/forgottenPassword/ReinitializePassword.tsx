import { t } from '@lingui/macro'
import { useNavigation, useRoute } from '@react-navigation/native'
import React, { useEffect, useState } from 'react'
import styled from 'styled-components/native'

import {
  isPasswordCorrect,
  PasswordSecurityRules,
} from 'features/auth/components/PasswordSecurityRules'
import { NavigateToHomeWithoutModalOptions } from 'features/navigation/helpers'
import { UseRouteType, UseNavigationType } from 'features/navigation/RootNavigator'
import { analytics } from 'libs/analytics'
import { _ } from 'libs/i18n'
import { BottomCardContentContainer } from 'ui/components/BottomCard'
import { BottomContentPage } from 'ui/components/BottomContentPage'
import { ButtonPrimary } from 'ui/components/buttons/ButtonPrimary'
import { InputError } from 'ui/components/inputs/InputError'
import { PasswordInput } from 'ui/components/inputs/PasswordInput'
import { ModalHeader } from 'ui/components/modals/ModalHeader'
import { SNACK_BAR_TIME_OUT, useSnackBarContext } from 'ui/components/snackBar/SnackBarContext'
import { Close } from 'ui/svg/icons/Close'
import { getSpacing, Spacer, Typo } from 'ui/theme'

import { useResetPasswordMutation } from '../mutations'

const MILLISECONDS_IN_A_SECOND = 1000

export const ReinitializePassword = () => {
  const { params } = useRoute<UseRouteType<'ReinitializePassword'>>()
  const { navigate } = useNavigation<UseNavigationType>()

  const { showSuccessSnackBar } = useSnackBarContext()

  const [password, setPassword] = useState('')
  const [shouldShowPasswordError] = useState(false)
  const [confirmedPassword, setConfirmedPassword] = useState('')
  const [shouldShowConfirmationError] = useState(false)

  const allowSubmission = isPasswordCorrect(password) && confirmedPassword === password

  const displayNotMatchingError = confirmedPassword.length > 0 && confirmedPassword !== password

  const { mutate: resetPassword, isLoading } = useResetPasswordMutation(() => {
    showSuccessSnackBar({
      message: _(t`Ton mot de passe a été modifié !`),
      timeout: SNACK_BAR_TIME_OUT,
    })
    analytics.logHasChangedPassword()
    navigate('Login')
  })

  function submitPassword() {
    resetPassword({
      newPassword: password,
      resetPasswordToken: params.token,
    })
  }

  function goToHomeWithoutModal() {
    navigate('Home', NavigateToHomeWithoutModalOptions)
  }

  useEffect(() => {
    if (params.expiration_timestamp * MILLISECONDS_IN_A_SECOND < new Date().getTime()) {
      navigate('Login')
    }
  }, [])

  return (
    <BottomContentPage>
      <ModalHeader
        title={_(t`Ton mot de passe`)}
        rightIcon={Close}
        onRightIconPress={goToHomeWithoutModal}
      />
      <BottomCardContentContainer>
        <Spacer.Column numberOfSpaces={6} />
        <StyledInput>
          <Typo.Body>{_(t`Nouveau mot de passe`)}</Typo.Body>
          <Spacer.Column numberOfSpaces={2} />
          <PasswordInput
            value={password}
            autoFocus
            onChangeText={setPassword}
            placeholder={_(/*i18n: password placeholder */ t`Ton mot de passe`)}
            isError={shouldShowPasswordError}
          />
        </StyledInput>
        <PasswordSecurityRules password={password} />
        <Spacer.Column numberOfSpaces={6} />
        <StyledInput>
          <Typo.Body>{_(t`Confirmer le mot de passe`)}</Typo.Body>
          <Spacer.Column numberOfSpaces={2} />
          <PasswordInput
            value={confirmedPassword}
            onChangeText={setConfirmedPassword}
            placeholder={_(/*i18n: password placeholder */ t`Confirmer le mot de passe`)}
            isError={shouldShowConfirmationError}
          />
        </StyledInput>
        <Spacer.Column numberOfSpaces={2} />
        <InputError
          visible={displayNotMatchingError}
          messageId="les mots de passe ne concordent pas"
          numberOfSpacesTop={0}
        />
        <Spacer.Column numberOfSpaces={6} />
        <ButtonPrimary
          title={_(t`Continuer`)}
          onPress={submitPassword}
          disabled={!allowSubmission}
          isLoading={isLoading}
        />
      </BottomCardContentContainer>
    </BottomContentPage>
  )
}

const StyledInput = styled.View({
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'flex-start',
  width: '100%',
  maxWidth: getSpacing(125),
})
