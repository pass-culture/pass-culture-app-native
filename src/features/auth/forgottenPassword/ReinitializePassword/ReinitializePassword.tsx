import { t } from '@lingui/macro'
import { useFocusEffect, useNavigation, useRoute } from '@react-navigation/native'
import React, { useCallback, useState } from 'react'
import { v4 as uuidv4 } from 'uuid'

import {
  isPasswordCorrect,
  PasswordSecurityRules,
} from 'features/auth/components/PasswordSecurityRules'
import { useResetPasswordMutation } from 'features/auth/mutations'
import { navigateToHome } from 'features/navigation/helpers'
import { UseNavigationType, UseRouteType } from 'features/navigation/RootNavigator'
import { analytics } from 'libs/analytics'
import { isTimestampExpired } from 'libs/dates'
import { BottomCardContentContainer } from 'ui/components/BottomCardContentContainer'
import { BottomContentPage } from 'ui/components/BottomContentPage'
import { ButtonPrimary } from 'ui/components/buttons/ButtonPrimary'
import { InputError } from 'ui/components/inputs/InputError'
import { PasswordInput } from 'ui/components/inputs/PasswordInput'
import { LoadingPage } from 'ui/components/LoadingPage'
import { ModalHeader } from 'ui/components/modals/ModalHeader'
import { SNACK_BAR_TIME_OUT, useSnackBarContext } from 'ui/components/snackBar/SnackBarContext'
import { Close } from 'ui/svg/icons/Close'
import { Spacer } from 'ui/theme'
import { Form } from 'ui/web/form/Form'

export const ReinitializePassword = () => {
  const route = useRoute<UseRouteType<'ReinitializePassword'>>()
  const navigation = useNavigation<UseNavigationType>()
  const { showSuccessSnackBar } = useSnackBarContext()

  const [isTimestampExpirationVerified, setIsTimestampExpirationVerified] = useState(false)
  const [password, setPassword] = useState('')
  const [confirmedPassword, setConfirmedPassword] = useState('')
  const passwordDescribedBy = uuidv4()
  const passwordErrorId = uuidv4()

  const allowSubmission = isPasswordCorrect(password) && confirmedPassword === password
  const displayNotMatchingError = confirmedPassword.length > 0 && confirmedPassword !== password

  useFocusEffect(
    useCallback(() => {
      setIsTimestampExpirationVerified(false)
      const { expiration_timestamp, email } = route.params
      if (isTimestampExpired(expiration_timestamp)) {
        navigation.replace('ResetPasswordExpiredLink', { email })
      } else {
        setIsTimestampExpirationVerified(true)
      }
    }, [route.params])
  )

  const { mutate: resetPassword, isLoading } = useResetPasswordMutation(() => {
    showSuccessSnackBar({
      message: t`Ton mot de passe a été modifié\u00a0!`,
      timeout: SNACK_BAR_TIME_OUT,
    })
    analytics.logHasChangedPassword('resetPassword')
    navigation.navigate('Login')
  })

  function submitPassword() {
    if (allowSubmission) {
      resetPassword({
        newPassword: password,
        resetPasswordToken: route.params.token,
      })
    }
  }

  if (!isTimestampExpirationVerified) {
    return <LoadingPage />
  }
  return (
    <BottomContentPage>
      <ModalHeader
        title={t`Ton mot de passe`}
        leftIconAccessibilityLabel={undefined}
        leftIcon={undefined}
        onLeftIconPress={undefined}
        rightIconAccessibilityLabel={t`Revenir à l'accueil`}
        rightIcon={Close}
        onRightIconPress={navigateToHome}
      />
      <BottomCardContentContainer>
        <Spacer.Column numberOfSpaces={6} />
        <Form.MaxWidth>
          <PasswordInput
            label={t`Nouveau mot de passe`}
            accessibilityDescribedBy={passwordDescribedBy}
            value={password}
            autoFocus
            onChangeText={setPassword}
            placeholder={t`Ton mot de passe`}
            onSubmitEditing={submitPassword}
            isRequiredField
          />
          <PasswordSecurityRules password={password} nativeID={passwordDescribedBy} />
          <Spacer.Column numberOfSpaces={6} />
          <PasswordInput
            label={t`Confirmer le mot de passe`}
            value={confirmedPassword}
            onChangeText={setConfirmedPassword}
            placeholder={t`Confirmer le mot de passe`}
            onSubmitEditing={submitPassword}
            isRequiredField
            accessibilityDescribedBy={passwordErrorId}
          />
          <Spacer.Column numberOfSpaces={2} />
          <InputError
            visible={displayNotMatchingError}
            messageId={t`Les mots de passe ne concordent pas`}
            numberOfSpacesTop={0}
            relatedInputId={passwordErrorId}
          />
          <Spacer.Column numberOfSpaces={6} />
          <ButtonPrimary
            wording={t`Continuer`}
            onPress={submitPassword}
            disabled={!allowSubmission || isLoading}
            isLoading={isLoading}
          />
        </Form.MaxWidth>
      </BottomCardContentContainer>
    </BottomContentPage>
  )
}
