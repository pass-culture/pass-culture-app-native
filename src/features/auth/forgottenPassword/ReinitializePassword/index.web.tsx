import loadable from '@loadable/component'
import React from 'react'

import { LoadingPage } from 'ui/components/LoadingPage'

export const ReinitializePassword = loadable(() => import('./ReinitializePassword'), {
  resolveComponent: (components) => components.ReinitializePassword,
  fallback: <LoadingPage />,
})
