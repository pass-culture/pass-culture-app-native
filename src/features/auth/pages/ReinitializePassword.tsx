import { t } from '@lingui/macro'
import { useNavigation, useRoute } from '@react-navigation/native'
import React, { useEffect, useState } from 'react'
import styled from 'styled-components/native'

import { UseNavigationType, UseRouteType } from 'features/navigation/RootNavigator'
import { _ } from 'libs/i18n'
import { BottomCard } from 'ui/components/BottomCard'
import { ButtonPrimary } from 'ui/components/buttons/ButtonPrimary'
import { PasswordInput } from 'ui/components/inputs/PasswordInput'
import { ModalHeader } from 'ui/components/modals/ModalHeader'
import { SafeContainer } from 'ui/components/SafeContainer'
import { Close } from 'ui/svg/icons/Close'
import { Warning } from 'ui/svg/icons/Warning'
import { ColorsEnum, getSpacing, Spacer, Typo } from 'ui/theme'

const MILLISECONDS_IN_A_SECOND = 1000

export const ReinitializePassword = () => {
  const {
    params: { expiration_date },
  } = useRoute<UseRouteType<'ReinitializePassword'>>()
  const navigation = useNavigation<UseNavigationType>()

  const [password, setPassword] = useState('')
  const [shouldShowPasswordError] = useState(false)
  const [confirmedPassword, setConfirmedPassword] = useState('')
  const [shouldShowConfirmationError] = useState(false)

  const arePasswordsMatching = confirmedPassword === password
  const allowSubmission = password.length > 0 && arePasswordsMatching

  function onClose() {
    navigation.navigate('Home', { shouldDisplayLoginModal: false })
  }
  function submitPassword() {
    // TODO: call submit function using the 'token' params
  }

  useEffect(() => {
    if (expiration_date * MILLISECONDS_IN_A_SECOND < new Date().getTime()) {
      navigation.navigate('Login')
    }
  }, [])

  return (
    <SafeContainer>
      <BottomCard>
        <ModalHeader title={_(t`Ton mot de passe`)} rightIcon={Close} onRightIconPress={onClose} />
        <Spacer.Column numberOfSpaces={6} />
        <StyledInput>
          <Typo.Body>{_(t`Nouveau mot de passe`)}</Typo.Body>
          <Spacer.Column numberOfSpaces={2} />
          <PasswordInput
            value={password}
            onChangeText={setPassword}
            placeholder={_(/*i18n: password placeholder */ t`Ton mot de passe`)}
            isError={shouldShowPasswordError}
            width="100%"
          />
        </StyledInput>
        <Spacer.Column numberOfSpaces={6} />
        <StyledInput>
          <Typo.Body>{_(t`Confirmer le mot de passe`)}</Typo.Body>
          <Spacer.Column numberOfSpaces={2} />
          <PasswordInput
            value={confirmedPassword}
            onChangeText={setConfirmedPassword}
            placeholder={_(/*i18n: password placeholder */ t`Ton mot de passe`)}
            isError={shouldShowConfirmationError}
            width="100%"
          />
        </StyledInput>
        <Spacer.Column numberOfSpaces={2} />
        {!arePasswordsMatching && (
          <ErrorLineContainer>
            <Warning size={24} color={ColorsEnum.ERROR} />
            <Typo.Caption color={ColorsEnum.ERROR}>
              {_(t`les mots de passe ne concordent pas`)}
            </Typo.Caption>
          </ErrorLineContainer>
        )}
        <Spacer.Column numberOfSpaces={6} />
        <ButtonPrimary
          title={_(t`Continuer`)}
          onPress={submitPassword}
          disabled={!allowSubmission}
        />
      </BottomCard>
    </SafeContainer>
  )
}

const StyledInput = styled.View({
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'flex-start',
  width: '100%',
  maxWidth: getSpacing(125),
})

const ErrorLineContainer = styled.View({
  flexDirection: 'row',
  alignItems: 'center',
  justifyContent: 'flex-start',
  width: '100%',
})
