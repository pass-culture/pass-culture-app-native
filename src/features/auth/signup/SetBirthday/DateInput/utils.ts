export interface DateValidation {
  isComplete: boolean
  isValid: boolean
  isDateAboveMin: boolean
  isDateBelowMax: boolean
}

export interface DateInputRef {
  clearFocuses: () => void
}
