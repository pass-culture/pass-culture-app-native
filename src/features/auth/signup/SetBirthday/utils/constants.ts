export const DEFAULT_YOUNGEST_AGE = 15
export const UNDER_YOUNGEST_AGE = DEFAULT_YOUNGEST_AGE - 1

export const MINIMUM_DATE = new Date('1900-01-01')
