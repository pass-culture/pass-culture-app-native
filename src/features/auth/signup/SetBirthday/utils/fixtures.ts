import { dateDiffInFullYears } from 'libs/dates'

export const CURRENT_DATE = new Date('2020-12-01T00:00:00.000Z')
export const ELIGIBLE_AGE_DATE = new Date('2003-12-01T00:00:00.000Z')
export const DEFAULT_SELECTED_DATE = new Date('2006-12-01T00:00:00.000Z')
export const MINIMUM_DATE = new Date('1900-01-01T00:00:00.000Z')
export const MAXIMUM_DATE = new Date('2003-12-01T00:00:00.000Z')
export const FUTUR_DATE = new Date('2022-12-01T00:00:00.000Z')
export const FIFTEEN_YEARS_OLD_FIRST_DAY_DATE = new Date('2005-12-01T00:00:00.000Z')
export const NOT_ELIGIBLE_YOUNGEST_AGE_DATE = new Date('2006-01-01T00:00:00.000Z')
export const AGE = dateDiffInFullYears(new Date(NOT_ELIGIBLE_YOUNGEST_AGE_DATE), CURRENT_DATE)
