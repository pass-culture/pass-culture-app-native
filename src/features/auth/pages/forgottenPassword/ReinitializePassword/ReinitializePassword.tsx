import { yupResolver } from '@hookform/resolvers/yup'
import { useFocusEffect, useNavigation, useRoute } from '@react-navigation/native'
import React, { useCallback, useEffect, useState } from 'react'
import { useForm } from 'react-hook-form'

import { useResetPasswordMutation } from 'features/auth/api/useResetPasswordMutation'
import { reinitializePasswordSchema } from 'features/auth/pages/forgottenPassword/ReinitializePassword/schema/reinitializePasswordSchema'
import { navigateToHome } from 'features/navigation/helpers'
import { UseNavigationType, UseRouteType } from 'features/navigation/RootNavigator/types'
import { analytics } from 'libs/analytics'
import { isTimestampExpired } from 'libs/dates'
import { PasswordInputController } from 'shared/forms/controllers/PasswordInputController'
import { ButtonPrimary } from 'ui/components/buttons/ButtonPrimary'
import { Form } from 'ui/components/Form'
import { RightButtonText } from 'ui/components/headers/RightButtonText'
import { LoadingPage } from 'ui/components/LoadingPage'
import { SNACK_BAR_TIME_OUT, useSnackBarContext } from 'ui/components/snackBar/SnackBarContext'
import { SecondaryPageWithBlurHeader } from 'ui/pages/SecondaryPageWithBlurHeader'
import { Spacer, Typo } from 'ui/theme'
import { getHeadingAttrs } from 'ui/theme/typographyAttrs/getHeadingAttrs'

type ReinitializePasswordFormData = {
  newPassword: string
  confirmedPassword: string
}

export const ReinitializePassword = () => {
  const defaultValues: ReinitializePasswordFormData = {
    newPassword: '',
    confirmedPassword: '',
  }

  const route = useRoute<UseRouteType<'ReinitializePassword'>>()
  const navigation = useNavigation<UseNavigationType>()
  const { showSuccessSnackBar } = useSnackBarContext()

  const [isTimestampExpirationVerified, setIsTimestampExpirationVerified] = useState(false)
  const {
    handleSubmit,
    control,
    formState: { isValid },
    watch,
    trigger,
  } = useForm<ReinitializePasswordFormData>({
    mode: 'onChange',
    defaultValues,
    resolver: yupResolver(reinitializePasswordSchema),
  })

  useFocusEffect(
    useCallback(() => {
      setIsTimestampExpirationVerified(false)
      const { expiration_timestamp, email } = route.params
      if (isTimestampExpired(expiration_timestamp)) {
        navigation.replace('ResetPasswordExpiredLink', { email })
      } else {
        setIsTimestampExpirationVerified(true)
      }
      // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [route.params])
  )

  // We use this useEffect in order to validate confirmedPassword when newPassword changes and matches
  const password = watch('newPassword')
  useEffect(() => {
    trigger('confirmedPassword')
  }, [password, trigger])

  const { mutate: resetPassword, isLoading } = useResetPasswordMutation(() => {
    showSuccessSnackBar({
      message: 'Ton mot de passe est modifiÃ©\u00a0!',
      timeout: SNACK_BAR_TIME_OUT,
    })
    analytics.logHasChangedPassword('resetPassword')
    navigation.navigate('Login')
  })

  const submitPassword = useCallback(
    ({ newPassword }: ReinitializePasswordFormData) => {
      if (isValid) {
        resetPassword({
          newPassword,
          resetPasswordToken: route.params.token,
        })
      }
    },
    [isValid, resetPassword, route.params.token]
  )

  if (!isTimestampExpirationVerified) {
    return <LoadingPage />
  }
  return (
    <SecondaryPageWithBlurHeader
      headerTitle="Nouveau mot de passe"
      RightButton={<RightButtonText onClose={navigateToHome} wording="Quitter" />}>
      <Spacer.Column numberOfSpaces={8} />
      <Typo.Title3 {...getHeadingAttrs(2)}>Choisis un nouveau mot de passe</Typo.Title3>
      <Spacer.Column numberOfSpaces={10} />
      <Form.MaxWidth>
        <PasswordInputController
          name="newPassword"
          label="Mot de passe"
          control={control}
          autoFocus
          isRequiredField
          withSecurityRules
          securityRulesAlwaysVisible
          onSubmitEditing={handleSubmit(submitPassword)}
        />
        <Spacer.Column numberOfSpaces={10} />
        <PasswordInputController
          name="confirmedPassword"
          label="Confirmer le mot de passe"
          placeholder="Confirmer le mot de passe"
          control={control}
          isRequiredField
          onSubmitEditing={handleSubmit(submitPassword)}
        />
        <Spacer.Column numberOfSpaces={10} />
        <ButtonPrimary
          wording="Continuer"
          onPress={handleSubmit(submitPassword)}
          disabled={!isValid || isLoading}
          isLoading={isLoading}
          accessibilityLabel="Valider le nouveau mot de passe et se connecter"
        />
      </Form.MaxWidth>
    </SecondaryPageWithBlurHeader>
  )
}
