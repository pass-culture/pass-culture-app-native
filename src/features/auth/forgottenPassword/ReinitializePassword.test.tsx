import { fireEvent, render } from '@testing-library/react-native'
import React from 'react'
import waitForExpect from 'wait-for-expect'

import { useRoute, navigate } from '__mocks__/@react-navigation/native'
import { analytics } from 'libs/analytics'
import { reactQueryProviderHOC } from 'tests/reactQueryProviderHOC'
import { superFlushWithAct } from 'tests/utils'
import { ColorsEnum } from 'ui/theme'

import { ReinitializePassword } from './ReinitializePassword'

describe('ReinitializePassword Page', () => {
  beforeAll(() => {
    useRoute.mockImplementation(() => ({
      params: {
        token: 'reerereskjlmkdlsf',
        expiration_timestamp: 45465546445,
      },
    }))
  })

  afterAll(() => {
    jest.resetAllMocks()
  })

  it('should enable the submit button when passwords are equals and filled and password is correct', async () => {
    const { getByPlaceholderText, getByTestId } = render(
      reactQueryProviderHOC(<ReinitializePassword />)
    )

    const passwordInput = getByPlaceholderText('Ton mot de passe')
    const confirmationInput = getByPlaceholderText('Confirmer le mot de passe')

    fireEvent.changeText(passwordInput, 'user@AZERTY123')
    fireEvent.changeText(confirmationInput, 'user@AZERTY123')

    // assuming there's only one button in this page
    const continueButton = getByTestId('button-container')

    await waitForExpect(async () => {
      const background = continueButton.props.style.backgroundColor
      expect(background).toEqual(ColorsEnum.PRIMARY)
    })
  })

  it('should display the matching error when the passwords dont match', async () => {
    const { getByPlaceholderText, getByText } = render(
      reactQueryProviderHOC(<ReinitializePassword />)
    )

    const passwordInput = getByPlaceholderText('Ton mot de passe')
    const confirmationInput = getByPlaceholderText('Confirmer le mot de passe')

    fireEvent.changeText(passwordInput, '123456')
    fireEvent.changeText(confirmationInput, '123456--')

    const notMatchingErrorText = getByText('les mots de passe ne concordent pas')

    await waitForExpect(async () => {
      const color = notMatchingErrorText.props.style[0].color
      expect(color).toEqual(ColorsEnum.ERROR)
    })
  })

  it('should validate PasswordSecurityRules when password is correct', async () => {
    const { toJSON, getByPlaceholderText } = render(reactQueryProviderHOC(<ReinitializePassword />))

    const notValidatedRulesSnapshot = toJSON()

    const passwordInput = getByPlaceholderText('Ton mot de passe')
    fireEvent.changeText(passwordInput, 'ABCDefgh1234!!!!')

    await waitForExpect(() => {
      const validatedRulesSnapshot = toJSON()
      expect(notValidatedRulesSnapshot).toMatchDiffSnapshot(validatedRulesSnapshot)
    })
  })

  it('should redirect to login page WHEN password is reset', async () => {
    const { getByText } = render(reactQueryProviderHOC(<ReinitializePassword />))
    fireEvent.press(getByText('Continuer'))
    await superFlushWithAct()
    expect(navigate).toBeCalledTimes(1)
    expect(analytics.logHasChangedPassword).toBeCalled()
    expect(navigate).toBeCalledWith('Login')
  })
})
