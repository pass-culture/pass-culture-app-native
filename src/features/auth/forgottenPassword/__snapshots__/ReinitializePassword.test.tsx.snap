// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReinitializePassword Page should validate PasswordSecurityRules when password is correct 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -272,11 +272,11 @@
                        \\"flexShrink\\": 1,
                      },
                    ]
                  }
                  underlineColorAndroid=\\"transparent\\"
-                 value=\\"\\"
+                 value=\\"ABCDefgh1234!!!!\\"
                />
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function onClick]}
@@ -327,21 +327,21 @@
                  ]
                }
              >
                <View
                  height={16}
-                 testID=\\"rule-icon-close\\"
+                 testID=\\"rule-icon-check\\"
                  width={16}
                >
-                 rule-icon-close-SVG-Mock
+                 rule-icon-check-SVG-Mock
                </View>
                <Text
-                 color=\\"#b60025\\"
+                 color=\\"#15884f\\"
                  style={
                    Array [
                      Object {
-                       \\"color\\": \\"#b60025\\",
+                       \\"color\\": \\"#15884f\\",
                        \\"fontFamily\\": \\"Montserrat-SemiBold\\",
                        \\"fontSize\\": 12,
                        \\"lineHeight\\": 16,
                        \\"paddingLeft\\": 4,
                      },
@@ -362,21 +362,21 @@
                  ]
                }
              >
                <View
                  height={16}
-                 testID=\\"rule-icon-close\\"
+                 testID=\\"rule-icon-check\\"
                  width={16}
                >
-                 rule-icon-close-SVG-Mock
+                 rule-icon-check-SVG-Mock
                </View>
                <Text
-                 color=\\"#b60025\\"
+                 color=\\"#15884f\\"
                  style={
                    Array [
                      Object {
-                       \\"color\\": \\"#b60025\\",
+                       \\"color\\": \\"#15884f\\",
                        \\"fontFamily\\": \\"Montserrat-SemiBold\\",
                        \\"fontSize\\": 12,
                        \\"lineHeight\\": 16,
                        \\"paddingLeft\\": 4,
                      },
@@ -397,21 +397,21 @@
                  ]
                }
              >
                <View
                  height={16}
-                 testID=\\"rule-icon-close\\"
+                 testID=\\"rule-icon-check\\"
                  width={16}
                >
-                 rule-icon-close-SVG-Mock
+                 rule-icon-check-SVG-Mock
                </View>
                <Text
-                 color=\\"#b60025\\"
+                 color=\\"#15884f\\"
                  style={
                    Array [
                      Object {
-                       \\"color\\": \\"#b60025\\",
+                       \\"color\\": \\"#15884f\\",
                        \\"fontFamily\\": \\"Montserrat-SemiBold\\",
                        \\"fontSize\\": 12,
                        \\"lineHeight\\": 16,
                        \\"paddingLeft\\": 4,
                      },
@@ -432,21 +432,21 @@
                  ]
                }
              >
                <View
                  height={16}
-                 testID=\\"rule-icon-close\\"
+                 testID=\\"rule-icon-check\\"
                  width={16}
                >
-                 rule-icon-close-SVG-Mock
+                 rule-icon-check-SVG-Mock
                </View>
                <Text
-                 color=\\"#b60025\\"
+                 color=\\"#15884f\\"
                  style={
                    Array [
                      Object {
-                       \\"color\\": \\"#b60025\\",
+                       \\"color\\": \\"#15884f\\",
                        \\"fontFamily\\": \\"Montserrat-SemiBold\\",
                        \\"fontSize\\": 12,
                        \\"lineHeight\\": 16,
                        \\"paddingLeft\\": 4,
                      },
@@ -467,21 +467,21 @@
                  ]
                }
              >
                <View
                  height={16}
-                 testID=\\"rule-icon-close\\"
+                 testID=\\"rule-icon-check\\"
                  width={16}
                >
-                 rule-icon-close-SVG-Mock
+                 rule-icon-check-SVG-Mock
                </View>
                <Text
-                 color=\\"#b60025\\"
+                 color=\\"#15884f\\"
                  style={
                    Array [
                      Object {
-                       \\"color\\": \\"#b60025\\",
+                       \\"color\\": \\"#15884f\\",
                        \\"fontFamily\\": \\"Montserrat-SemiBold\\",
                        \\"fontSize\\": 12,
                        \\"lineHeight\\": 16,
                        \\"paddingLeft\\": 4,
                      },"
`;
